packages/di/src/common/utils/providerBuilder.ts

```
@@ -49,11 +53,11 @@ export function providerBuilder<Provider, Picked extends keyof Provider>(props:
         };
       },
       {
-        factory(factory: any) {
+        factory(factory: Function) {
           provider.useFactory = factory;
           return this;
         },
-        asyncFactory(asyncFactory: any) {
+        asyncFactory(asyncFactory: Function) {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)